### PR TITLE
feat: add multiple components to view and edit all external services

### PR DESCRIPTION
**Pull request by luckyman20**
## Description
Designed the screens to view and edit all the external services (SMS, Email, Notification and Amazon S3).

## Related issues and discussion
#182 and #183 

## Screenshots, if any
view_notification
Notification

view_sms
SMS

view_email
Email

view_amazon_s3
Amazon S3

selection_019
Edit Notification

selection_018
Edit Email

selection_017
Edit SMS

selection_016
Edit Amazon S3

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/system/external-services/amazon-s3/amazon-s3.component.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<div class="container m-b-20" fxLayout="row" fxLayoutAlign="end" fxLayoutGap="20px">
+    <button mat-raised-button color="primary" [routerLink]="['edit']">
+      <fa-icon icon="edit"></fa-icon>&nbsp;&nbsp;
+      Edit
+    </button>
+  </div>
+
+<div class="container">
+
+  <div class="mat-container-z8">
```
Comment (abhaychawla) [inline src/app/system/external-services/amazon-s3/amazon-s3.component.html]: The class is `mat-elevation-z8`. Make changes in all the files necessary. Also, a single div can be used here.

File: src/app/system/external-services/amazon-s3/edit-amazon-s3/edit-amazon-s3.component.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<div class="container">
+
+  <mat-card>
+
+    <form [formGroup]="editAmazonS3ConfigurationForm" (ngSubmit)="submit()">
```
Comment (abhaychawla) [inline src/app/system/external-services/amazon-s3/edit-amazon-s3/edit-amazon-s3.component.html]:
```suggestion
    <form [formGroup]="amazonS3ConfigurationForm" (ngSubmit)="submit()">
```
No need to mention edit in the form name. Make changes in all files necessary.

File: src/app/system/system-routing.module.ts [hunk 3]
```diff
@@ -27,42 +35,135 @@ import { ManageSurveysResolver } from './manage-surveys/manage-surveys.resolver'
...
+import { SMSConfigurationResolver } from './external-services/sms/sms.resolver';
+import { NotificationConfigurationResolver } from './external-services/notification/notification.resolver';
 
 const routes: Routes = [
   Route.withShell([
     {
       path: 'system',
       data: { title: extract('System'), breadcrumb: 'System' },
       children: [
-      {
```
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]: Remove changes to indentation from this PR.

File: src/app/system/system-routing.module.ts [hunk 4]
```diff
@@ -27,42 +35,135 @@ import { ManageSurveysResolver } from './manage-surveys/manage-surveys.resolver'
...
+                  component: EditAmazonS3Component,
+                  data: {title: extract('Edit Amazon S3 Configuration'), breadcrumb: 'Edit'},
+                  resolve: {
+                    amazonS3Configuration: AmazonS3ConfigurationResolver
+                  }
+                }
+              ]
+            },
+            {
+              path: 'SMTP',
```
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]:
```suggestion
              path: 'email',
```

File: src/app/system/system-routing.module.ts [hunk 5]
```diff
@@ -27,42 +35,135 @@ import { ManageSurveysResolver } from './manage-surveys/manage-surveys.resolver'
...
+                  component: EditEmailComponent,
+                  data: { title: extract('Edit Email Configuration'), breadcrumb: 'Edit' },
+                  resolve: {
+                    emailConfiguration: EmailConfigurationResolver
+                  }
+                }
+              ]
+            },
+            {
+              path: 'SMS',
```
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]:
```suggestion
              path: 'sms',
```

File: src/app/system/system-routing.module.ts [hunk 6]
```diff
@@ -27,42 +35,135 @@ import { ManageSurveysResolver } from './manage-surveys/manage-surveys.resolver'
...
+                  data: { title: extract('Edit SMS Configuration'), breadcrumb: 'Edit' },
+                  component: EditSMSComponent,
+                  resolve: {
+                    smsConfiguration: SMSConfigurationResolver
+                  }
+                }
+              ]
+            },
+            {
+              path: 'NOTIFICATION',
```
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]:
```suggestion
              path: 'notification',
```

File: src/app/system/system-routing.module.ts [hunk 7]
```diff
@@ -27,42 +35,135 @@ import { ManageSurveysResolver } from './manage-surveys/manage-surveys.resolver'
...
-      },
-      {
+          data: { title: extract('External Services'), breadcrumb: 'External Services' },
+          children: [
+            {
+              path: '',
+              component: ExternalServicesComponent
+            },
+            {
+              path: 'S3',
```
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]:
```suggestion
              path: 's3-amazon',
```

File: src/app/system/system-routing.module.ts (hunk 4 above)
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]:
```suggestion
              path: 'email',
```

File: src/app/system/system-routing.module.ts (hunk 5 above)
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]:
```suggestion
              path: 'sms',
```

File: src/app/system/system-routing.module.ts (hunk 6 above)
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]:
```suggestion
              path: 'notification',
```

File: src/app/system/system-routing.module.ts (hunk 7 above)
Comment (luckyman20) [reply src/app/system/system-routing.module.ts]: I think it should be `amazon-s3` then.

Comment (abhaychawla) [reply src/app/system/system-routing.module.ts]: Yes alright.

File: src/app/system/external-services/amazon-s3/amazon-s3.component.html [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  </button>
+</div>
+
+<div class="container mat-elevation-z8">
+
+  <table mat-table [dataSource]="dataSource" matSort>
+
+    <ng-container matColumnDef="name">
+      <th mat-header-cell *matHeaderCellDef mat-sort-header> Name </th>
+      <td mat-cell *matCellDef="let configuration"> {{ configuration.name }}&nbsp;&nbsp;
```
Comment (abhaychawla) [inline src/app/system/external-services/amazon-s3/amazon-s3.component.html]:
```suggestion
      <td mat-cell *matCellDef="let configuration"> {{ configuration.name }} </td>
```
Update for all cases in all files.

File: src/app/system/external-services/email/email.component.html [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+      <td mat-cell *matCellDef="let configuration"> {{ configuration.value }}&nbsp;&nbsp;
+      </td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
+
+  </table>
+
+</div>
```
Comment (abhaychawla) [inline src/app/system/external-services/email/email.component.html]: Add a line at the end of file.

File: src/app/system/external-services/amazon-s3/edit-amazon-s3/edit-amazon-s3.component.ts [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+  /**
+   * Creates the Amazon S3 configuration form.
+   */
+  ngOnInit() {
+    this.createEditAmazonS3ConfigurationForm();
+  }
+
+  /**
+   * Creates the edit Amazon S3 configuration form.
```
Comment (abhaychawla) [inline src/app/system/external-services/amazon-s3/edit-amazon-s3/edit-amazon-s3.component.ts]:
```suggestion
   * Creates Amazon S3 configuration form.
```
No need to mention edit, make changes in all files necessary.

File: src/app/system/external-services/amazon-s3/edit-amazon-s3/edit-amazon-s3.component.ts [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+   */
+  createEditAmazonS3ConfigurationForm() {
```
Comment (abhaychawla) [inline src/app/system/external-services/amazon-s3/edit-amazon-s3/edit-amazon-s3.component.ts]:
```suggestion
  createAmazonS3ConfigurationForm() {
```
Same as above.

File: src/app/system/external-services/notification/notification.component.html [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+      <td mat-cell *matCellDef="let configuration"> {{ configuration.value }}&nbsp;&nbsp;
+      </td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
+
+  </table>
+
+</div>
```
Comment (abhaychawla) [inline src/app/system/external-services/notification/notification.component.html]: Add a line at the end of file.

File: src/app/system/external-services/sms/sms.component.html [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+      <td mat-cell *matCellDef="let configuration"> {{ configuration.value }}&nbsp;&nbsp;
+      </td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
+
+  </table>
+
+</div>
```
Comment (abhaychawla) [inline src/app/system/external-services/sms/sms.component.html]: Add a line at the end of file.

File: src/app/system/system-routing.module.ts [hunk 14]
```diff
@@ -59,10 +71,99 @@ const routes: Routes = [
       },
       {
           path: 'external-services',
-          component: ExternalServicesComponent,
```
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]: The commit should show only the removal of this line. Please remove any space or indentation change which makes it remove 4 lines instead.

File: src/app/system/system.service.ts [hunk 15]
```diff
@@ -99,4 +99,21 @@ export class SystemService {
     return this.http.put(`/configurations/${configurationId}`, configuration);
   }
 
+  /**
+   * @param {string} externalConfigurationName External Configuration Name.
+   * @returns {Observable<any>}
```
Comment (abhaychawla) [inline src/app/system/system.service.ts]: Mention what it returns in the comment for `get` functions.

File: src/app/system/system-routing.module.ts (hunk 14 above)
Comment (luckyman20) [reply src/app/system/system-routing.module.ts]: There are no space or indentation change in that, it is restructured that's why it showing that. All 4 lines are changed so it shows that they are removed.

**Pull request merged**